### PR TITLE
Cleanup/simplification

### PR DESCRIPTION
**Pull request by Bond-009**
* Removed useless copies/allocations
* Reduced unneeded complexity

### CODE REVIEW

**Comment by cvium** [comment]
Some conflicts with #1010

File: MediaBrowser.Controller/Entities/BaseItem.cs [hunk 1]
```diff
@@ -2452,10 +2462,8 @@ protected List<FileSystemMetadata> GetLocalMetadataFilesToDelete()
             }
 
             var filename = System.IO.Path.GetFileNameWithoutExtension(Path);
-            var extensions = new List<string> { ".nfo", ".xml", ".srt", ".vtt", ".sub", ".idx", ".txt", ".edl", ".bif", ".smi", ".ttml" };
-            extensions.AddRange(SupportedImageExtensions);
```
Comment (JustAMan) [inline MediaBrowser.Controller/Entities/BaseItem.cs]: why this is no longer added to `extensions`?

Comment (Bond-009) [reply MediaBrowser.Controller/Entities/BaseItem.cs]: It is

**Comment by ploughpuff** [comment]
Seems this change is causing an exception:
```

System.TypeInitializationException
  HResult=0x80131534
  Message=The type initializer for 'MediaBrowser.Controller.Entities.BaseItem' threw an exception.
  Source=MediaBrowser.Controller
  StackTrace:
   at MediaBrowser.Controller.Entities.BaseItem.set_Logger(ILogger value) in C:\git_repos\jellyfin\MediaBrowser.Controller\Entities\BaseItem.cs:line 550
   at Emby.Server.Implementations.ApplicationHost.SetStaticProperties() in C:\git_repos\jellyfin\Emby.Server.Implementations\ApplicationHost.cs:line 1020
   at Emby.Server.Implementations.ApplicationHost.<RegisterResources>d__254.MoveNext() in C:\git_repos\jellyfin\Emby.Server.Implementations\ApplicationHost.cs:line 897
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Emby.Server.Implementations.ApplicationHost.<Init>d__246.MoveNext() in C:\git_repos\jellyfin\Emby.Server.Implementations\ApplicationHost.cs:line 617

Inner Exception 1:
ArgumentNullException: Value cannot be null.
Arg_ParamName_Name

```

**Comment by ploughpuff** [comment]
The exception issue I mentioned is being addressed in PR #1071

**Pull request merged**